packages/experimental-identity-and-auth/src/index.ts

```diff
@@ -0,0 +1,108 @@
+import { HttpRequest, Identity, IdentityProvider } from "@smithy/types";
+
+/**
+ * Interface to get an IdentityProvider for a specified HttpAuthScheme
+ * @internal
+ */
+export interface IdentityProviderConfiguration {
+  /**
+   * Get the IdentityProvider for a specified HttpAuthScheme.
+   * @param schemeId schemeId of the HttpAuthScheme
+   * @returns IdentityProvider or undefined if HttpAuthScheme is not found
+   */
+  getIdentityProvider(schemeId: string): IdentityProvider<Identity> | undefined;
+
+  /**
+   * Gets the configured HttpAuthSchemes.
+   * @returns all configured HttpAuthSchemes
+   */
+  getAuthSchemes(): Map<string, HttpAuthScheme>;
```

Comment:
are these keys known ahead of time? perhaps you could use a union that you can expand later.

Reply:
These keys are known at codegen time, so I think it'd be difficult pass that information into this TS implementation.

Fo simplicity, we can create a type alias in case we want to expand auto-complete later.

```typescript
type HttpAuthSchemeId = string;
```